dl1_data_handler/reader.py

```diff
@@ -1,2028 +1,1202 @@
+        help=(
+            "Minimum number of telescopes required globally after ``TableQualityQuery``. "
+            "Events with fewer telescopes will be filtered out completely. "
+            "Requires mode to be ``stereo``."
+        ),
+    ).tag(config=True)
+
+    min_telescopes_of_type = IntTelescopeParameter(
+        default_value=0,
+        help=(
+            "Minimum number of telescopes required for a specific type after ``TableQualityQuery``. "
+            "In events with fewer telescopes of that type, "
+            "those telescopes will be removed from the array event. "
+            "This might result in the event not fulfilling ``min_telescopes`` anymore "
+            "and thus being filtered completely. "
+            "Requires mode to be ``stereo``. "
+        ),
+    ).tag(config=True)
+
     def __init__(
         self,
-        file_list,
-        example_identifiers_file=None,
-        mode="mono",
-        selected_telescope_types=None,
-        selected_telescope_ids=None,
-        multiplicity_selection=None,
-        event_selection=None,
-        parameter_selection=None,
-        shuffle=False,
-        seed=None,
-        trigger_settings=None,
-        waveform_settings=None,
-        image_settings=None,
-        mapping_settings=None,
-        parameter_settings=None,
-        subarray_info=None,
-        event_info=None,
-        transforms=None,
-        validate_processor=False,
+        config=None,
+        parent=None,
+        **kwargs,
     ):
 
-        # Construct dict of filename:file_handle pairs
+        super().__init__(config=config, parent=parent, **kwargs)
+
+        # Initialize the Table data quality query
+        self.quality_query = TableQualityQuery(parent=self)
+
+        # Construct dict of filename:file_handle pairs of an ordered file list
         self.files = OrderedDict()
-        # Order the file_list
-        file_list = np.sort(file_list)
-        for filename in file_list:
+        file_list = (
+            self.signal_input_files
+            if self.bkg_input_files is None
+            else self.signal_input_files + self.bkg_input_files
+        )
+        for filename in np.sort(file_list):
             with lock:
                 self.files[filename] = tables.open_file(filename, mode="r")
-        first_file = list(self.files)[0]
-
-        # Save the user attributes and useful information retrieved from the first file
-        self._v_attrs = self.files[first_file].root._v_attrs
-        self.subarray_layout = self.files[
-            first_file
-        ].root.configuration.instrument.subarray.layout
-        self.tel_ids = self.subarray_layout.cols._f_col("tel_id")
+        self.first_file = list(self.files)[0]
+        # Save the user attributes and useful information retrieved from the first file as a reference
+        self._v_attrs = self.files[self.first_file].root._v_attrs
         self.process_type = self._v_attrs["CTA PROCESS TYPE"]
```

Comment:
Here you can apply Enum for "CTA PROCESS TYPE" and others . More easy to change in the future if the name change.